robolectric/src/main/java/org/robolectric/internal/ParallelUniverse.java

```diff
@@ -139,28 +142,16 @@ public void setUpApplicationState(Method method, TestLifecycle testLifecycle, Re
 
   @Override
   public void tearDownApplication() {
+    // We would like to null this out (which would call reset) but that can kill async threads
```

Comment:
What kind of async threads? Unit tests shouldn't be running arbitrary threads and all "threads" should really be funneled over the Scheduler. I'd like to force run everything on the scheduler before the tests exit in a future PR, but I think that, like nulling this out would also break rouge tests.